beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -1406,10 +1400,7 @@ impl<E: EthSpec> Network<E> {
         let peer_id = event.peer_id;
 
         // Do not permit Inbound events from peers that are being disconnected, or RPC requests.
-        if !self.peer_manager().is_connected(&peer_id)
-            && (matches!(event.event, HandlerEvent::Err(HandlerErr::Inbound { .. }))
```

Comment:
Why do we not need these conditions anymore?

Reply:
cause `HandlerEvent` is no longer exported in `RpcMessage::event` it's now `Result<RPCReceived<Id, E>, HandlerErr<Id>>`

Reply:
But the reason we had the extra condition iirc is that we wanted to generate a RpcFailed event for `HandlerErr::Outbound` to send to sync and allow sync to remove the peer from all mappings

Reply:
Yeah, nice catch

Reply:
ah right! Thanks Pawan, that makes sense yeah, submmited https://github.com/sigp/lighthouse/pull/6408/commits/5fa8c80fa2060ab4162c7cd662dc9e888102efd7